awscli/clidriver.py

```diff
@@ -379,15 +366,17 @@ def __call__(self, args, parsed_globals):
 
     def _create_command_table(self):
         command_table = OrderedDict()
-        service_object = self._get_service_object()
-        for operation_object in service_object.operations:
-            cli_name = xform_name(operation_object.name, '-')
+        service_model = self._get_service_model()
```

Comment:
After looking into this further, I'm not sure if we can remove the service_model entirely.  Even if we type check for a ServiceCommand, we need to know what the service name is in order to retrieve the correct service model.  However, because a command can be renamed, we can't rely on the command name being the same as what we use to `get_sevice_model`.

Have I overlooked anything?